contracts/NonfungiblePositionManager.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./base/PeripheryImmutableState.sol";
+import "./base/PeripheryValidation.sol";
+import "./interfaces/INonfungiblePositionManager.sol";
+import {TickMath} from "@uniswap/v4-core/contracts/libraries/TickMath.sol";
+import {Pool} from "@uniswap/v4-core/contracts/libraries/Pool.sol";
+import {Position} from "@uniswap/v4-core/contracts/libraries/Position.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {LiquidityAmounts} from "./libraries/LiquidityAmounts.sol";
+import {IPoolManager, PoolManager} from "@uniswap/v4-core/contracts/PoolManager.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/contracts/types/PoolId.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+
+contract NonfungiblePositionManager is
+    ERC721,
+    PeripheryImmutableState,
+    INonfungiblePositionManager,
+    PeripheryValidation
+{
+    using PoolIdLibrary for PoolKey;
+
+    /// @dev The ID of the next token that will be minted. Skips 0
+    uint256 private _nextId = 1;
+
+    enum CallbackDataType {
+        Mint,
+        IncreaseLiquidity
+    }
+
+    struct CallbackData {
+        address sender;
+        CallbackDataType callbackDataType;
+        MintParams params;
+    }
+
+    // details about the uniswap position
+    struct TokenIdPosition {
+        PoolKey poolKey;
+        // the tick range of the position
+        int24 tickLower;
+        int24 tickUpper;
+        // the liquidity of the position
+        uint128 liquidity;
+        // the fee growth of the aggregate position as of the last action on the individual position
+        uint256 feeGrowthInside0LastX128;
+        uint256 feeGrowthInside1LastX128;
+        // how many uncollected tokens are owed to the position, as of the last computation
+        uint128 tokensOwed0;
+        uint128 tokensOwed1;
+    }
+
+    struct AddLiquidityParams {
+        PoolKey poolKey;
+        int24 tickLower;
+        int24 tickUpper;
+        uint256 amount0Desired;
+        uint256 amount1Desired;
+        uint256 amount0Min;
+        uint256 amount1Min;
+    }
+
+    /// @dev The token ID position data
+    mapping(uint256 => TokenIdPosition) public positions;
+
+    constructor(PoolManager _poolManager, address _WETH9)
```

Comment:
any reason why this isn't IPoolManager?

Reply:
Agreed, putting on TO DO list to change.